frappe/public/js/frappe/form/controls/base_control.js

```diff
@@ -94,6 +94,46 @@ frappe.ui.form.Control = Class.extend({
 			&& this.$wrapper.toggleClass("hide-control", this.disp_status=="None")
 			&& this.refresh_input
 			&& this.refresh_input();
+		
+		var value = this.get_value();
+
+		// Disable translation non-string fields or special string fields
+		if (!value
+			|| !frappe.user_roles.includes('Translator')
```

Comment:
instead of hard-coding, we can check on basis of permission for "Translation"

Reply:
Also lets also tag fields as "translatable" via docfield so it does not show up for every field